static/js/cf-frontend.js

```diff
@@ -346,15 +355,31 @@ function selectedCheckerOnChange() {
 
 
 }
+
+function linkGen(){
+  var checker_value = $("#type_system").val();
+  var typeURL = encodeURI(checker_value);
+  typeURL = "#type=" + typeURL;
+
+  var input = pyInputGetValue(); 
+  var inputURL = encodeURI(input);
+  inputURL = "&code=" + inputURL;
+
+  var curUrl = window.location.host + '/';
+  document.getElementById("link").value = (curUrl + typeURL + inputURL);
+}
+
+
 /*====general functions====*/
 // run at the END so that everything else can be initialized first
 function genericOptFrontendReady() {
   // be friendly to the browser's forward and back buttons
   // thanks to http://benalman.com/projects/jquery-bbq-plugin/
   $(window).bind("hashchange", function(e) {
-    // if you've got some preseeded code, then parse the entire query
-    // string from scratch just like a page reload
-    if ($.bbq.getState('code')) {
+    //preceeded code or type exists, initalize the page accordingly
```

Comment:
Code comment is more like a documentation to others, which describe your idea behind the implementation, especially when your idea cannot easily see through reading the code.

Currently this code comment is more like a side note for yourself, so please improve it. Maybe some code comment like:

"Parsing url hash-tag parameters if we have `code` or `typeSystem` updated/encoded in bbq states. Currently the encoded url will contains both `code` and `typeSystem`. Here the `or` condition is for future extension, i.e. when only update/encoding `code` or `typeSystem `, we will also parse url hash-tag parameters here."